.github/workflows/holiday-generator/main.js

```diff
@@ -0,0 +1,125 @@
+import { createHash } from "node:crypto";
+import { writeFile } from "node:fs/promises";
+import { join } from "node:path";
+import { promisify } from "node:util";
+
+import Holidays from "date-holidays";
+import { createEvents as icsCreateEvents } from "ics";
+
+import { COUNTRIES, END_YEAR, FIXED_DATE_START_YEAR, ICS_PATH, SHOULD_LOG, START_YEAR, TYPE_WHITELIST } from "./config.js";
+
+// converting createEvents from ics from function with callback to async function for easier usage
+const createEvents = promisify(icsCreateEvents);
+
+/**
+ * Log info to console
+ * @param {*} toLog
+ */
+function log(toLog) {
+    if (SHOULD_LOG) {
+        console.log(toLog);
+    }
+}
+
+/**
+ * Get events for country given by code
+ * @param {string} countryCode
+ * @returns
+ */
+function getEvents(countryCode) {
+    const generator = new Holidays(countryCode);
+    const events = [];
+    for (let i = START_YEAR; i <= END_YEAR; i++) {
+        events.push(...generator.getHolidays(i).filter((x) => TYPE_WHITELIST.includes(x.type)));
+    }
+    return events;
+}
+
+/**
+ * Generates reproducible ID for holiday
+ * @param {string} countryCode
+ * @param {string} date
+ * @param {string} rule
+ * @returns
+ */
+function generateUid(countryCode, date, rule) {
+    const hashGen = createHash("sha256");
+    hashGen.update(`${countryCode},${date},${rule}`);
+    return hashGen.digest("hex");
+}
+
+/**
+ * Convert JS Date object to ics.DateTime array
+ * @param {Date} date
+ * @returns
+ */
+function getDateArray(date) {
+    return [date.getFullYear(), date.getMonth() + 1, date.getDate()];
+}
+
+/**
+ * Checks if holiday is a fixed-date holiday.
+ * Regex based on https://github.com/commenthol/date-holidays/blob/master/docs/specification.md#fixed-date
+ * @param {string} rule
+ * @returns
+ */
+function isFixedDate(rule) {
+    return /^\d\d-\d\d$/.test(rule);
+}
+
+/**
+ * Generate ical file from given set of events
+ * @param {ReturnType<getEvents>} events
+ * @param {string} countryCode
+ * @returns {Promise<string>}
+ */
+async function generateIcal(events, countryCode) {
+    const eventsMap = new Map();
+    events.forEach((x) => {
+        if (isFixedDate(x.rule)) {
+            const uid = generateUid(countryCode, "", x.rule);
+            if (!eventsMap.has(uid)) {
+                const yearDiff = x.end.getFullYear() - x.start.getFullYear();
+                x.start.setFullYear(FIXED_DATE_START_YEAR);
+                x.end.setFullYear(FIXED_DATE_START_YEAR + yearDiff);
+                eventsMap.set(uid, {
+                    title: x.name,
+                    uid,
+                    start: getDateArray(x.start),
+                    end: getDateArray(x.end),
+                    recurrenceRule: "FREQ=YEARLY",
```

Comment:
We should probably append us `FREQ=YEARLY;BYMONTH=$month;BYMONTHDAY=$dayOfMonth`

Reply:
I've checked current ICS files and there are either `RRULE:FREQ=YEARLY` or `RRULE:FREQ=YEARLY;INTERVAL=1`.

Reply:
I can modify this way, but I don't know if it's necessary. The current way seems to work fine, but if there's any advantage in specifying exact month and day, I can change it. 

I'm rather thinking about specifying `INTERVAL=1` every time because after checking random ICS files in Calendar, only `poland.ics` has no interval defined (but it's adding dates correctly).

Reply:
There's only one advantage that it's more explicit but I tested importing a couple of events with `FREQ=YEARLY` and it seems to work fine so we can leave it as it is. It's not like the ICS files will be used outside the app.

The `FREQ=YEARLY;INTERVAL=X` means the event should repeat every 1 year. Not needed as far as I know.